### ISSUE TITLE
Xourna++ as All-in-One (PDF Writer, Reader, Form Filler, PDF Manipulator) and most popular / versatile opensource PDF tool

### ISSUE THREAD

**Issue by osering** [xournalpp/xournalpp#5914]
### Is Your Feature Request Related to a Problem?

**Problem**:  Xournal++ is niche product, known by comparatively small number of Linux users. Even if someone finds out and loves Xournal++ for its cool features and rich editing capabilities, he/she needs to install/use in parallel some PDF Reader as well, as Xournal++ right now is not capable being used as even low-end PDF reader. This also should be mentioned that every Linux distribution puts some PDF viewer as installed by default (be it Evince, Okular, ...), therefore being decent PDF viewer is important.
On the other hand - there not so much additions necessary to make Xournal++ decent PDF Reader (any PDF Reader to become PDF Writer takes much more effort/resources and devotion). So with less input there is possibility of acquiring more users from competitors.

If there would be PDF form filling capabilities, like Firefox has - that would be marvelous addition (as there is vacuum in Linux space PDF tools with this capability). Some PDF Readers offer Digital Signing as well, to cover most popular usecases.

**Vision**: Xournal++ is one stop shop for Linux PDF users & includes 3 global modes: 1) Presenter/Watcher/Reader, 2) Editor/Writer, 3) Form Filler (preferably, each with different/relevant Menubar items) and it becomes the default app for most PDF users and most Linux distributions, like Arch, Debian, Fedora, Ubuntu...
Xournal++ become best and most versatile opensource product for PDF needs (Linux, BSD), hitting headlines and even getting good share of closedsource lovers form Windows and Mac ecosystem. 

P.S. If to look globally at PDF Tools, at the beginning PDF consumer apps was called readers (as most of PDF content was text) then now it would be more appropriate call these PDF presenters/watchers (instead of readers). Editors like Xournal++ actually is just Layer editors and as addition - FormFillers and PDF Arranger (restructure PDFs - add (blank or with picture, like PNG, JPG), remove, rearrange pages, add OCR layer to PDF https://github.com/ocrmypdf/OCRmyPDF, add audio, video, 3D), PDF Converters - convert PDF to/from various formats, like eBook (ePub, FB2, Mobi), XPS, DjVu, CHM, Comic Book (CBZ and CBR)  formats)). PDF Exporters  like OCR to TTS,. Therefore (key) shortcuts could be (W)atch / (P)resentation, (L)ayer Editor and (F)orm Filler, (A)rranger, (C)onverter, (E)xporter to switch between.
Here new capabilities are exposed as vector for development: https://helpx.adobe.com/uk/acrobat/using/rich-media.html

### Feature Description

Introduce so called PDF Watch/Present (Read) and PDF FormFiller modes (in addition to present Layer editor mode)
Feature same as Youtube has, when pressing key (F) to enter Full Screen mode, which gives user similar experience as using some PDF reader and therefore getting wider acceptance in more usecases. Or alternatively key (R) as for Readers Mode (this can overlap with page rotate shortcut) or (P) for Presentation mode.

Right now someone has to press 4 buttons F9, F10, F11, F12 to (toggle) get rid of all screen blocks or F5 + F9, but it is impossible to remove them all at once, to have immersive watching/presentation experience, which reader desires.

This Present/Watch/Read mode could be called Fullscreen (or Fullscreen Watch) mode if we want to be consistent with Youtube "de facto" standard which has lot of users and usage of (F) key would be more use to as well. It would require to rename current Fullscreen mode (F11) to something else, like Fulscreen LayerEdit (or Header-less mode), contrary to Fullscreen Watch/Presentation. 
In this mode leave just keystrokes available (mouse, touchpad, keyboard) for navigation and small floating window for showing current page / zoom%  (activated during scroll or with some Ctrl+ shortcut ) and letting input page number where jump to and zoom%. Available Help (H or Ctrl+H), showing all shortcuts. 

Watching/Presentation mode should be available in different scales/zooms: Fit height (H or Ctrl+H), Fit width (W or Ctrl+W), 100% (0 or Ctrl+0) and change it with just Ctrl +  (-) and (=) buttons WITHOUT Shift usage (like Firefox scaling works). Right now to Zoom in, one have to press 3 buttons Ctrl + Shift + (=/+), but to Zoom Out - just 2 (no Shift needed) Ctrl + (-/_). This inconsistency is annoying.
Inverted background (to black/gray or user selected) with Ctrl + (i) could give extra value as PDF reader (to lessen strain to eyes white background provides).
Bookmarking capability should be added (which most of PDF Readers has).
For Watch/Presentation mode Firefox PDF viewer / QPDFview / Atril document viewer (or some other) could be taken as  example (including shortcuts/views/bookmark functions implementation, choosing which model in which area to follow).

Mode switch should be persistent (user can set in settings default one) or choose to remember last mode&view, when app being opened again.

### Alternative Solutions

Use proprietary Adobe solutions and live in "golden cage" of this shitty company. 

### Additional Context

In wider context: with well developed Watch/Presentation/Read mode Xournal++ could immensely  widen its usage and become one of the best PDF Readers in addition being one of the best PDF Layer Editors. Next step - Form Filler, as well as audio, video, 3D Embeder,  to become all rounder, so users do not have to look around for other tools for their needs.
I am aware that such revamp can introduce massive contemplation and brainstorm about optimal shortcuts, views, but in my view it's worth it to get in higher class and become the game-changer and Adobe challenger.

**Comment by MMachado05** [comment]
I feel like this is an Issue made more out of an undefined ideological desire rather than a concrete solution to a well-experienced problem. Like...

> **Alternative Solutions**
>
> Use proprietary Adobe solutions and live in "golden cage" of this shitty company.

This isn't "solution" language, this is just woefully observing the world as it is. Not an unproductive thing to do in one's life, but certainly not productive in a code repository.

Furthermore, a lot of things here are either a mish-mash of various feature requests that likely should be filed as their own separate issues (such as your tool/view toggle request), or such high-reaching goals that you're pretty much asking for the dev team to create an entire new project (implementing a file-format converter and PDF interaction are such gargantuan tasks there there are projects that literally just do [only](https://flathub.org/apps/search?q=file+convert) [that](https://flathub.org/apps/search?q=pdf)).

And I say this as someone who probably shares your sentiment of wanting FOSS to be the norm. But we get there by creating good products, with intuitive solutions to common problems, good UIs, strong funding channels, and well-established communities. Not by clogging the repo's Issues with half-rants.

Also, and I hate to be a wet towel here, but you're making "total market domination" out to be a *lot* easier than it is. It took Blender *years* of community-building and the collection of hundres of *thousands* of dollars in monthly donations to get to where it is today. It took MuseScore a *huge* rework, a corporate restructuring, and influencer goodwill for it to do the same. We're *maybe* seeing something similar happen to Godot right now. If we really wanna see projects like these get to those points, we can faff around unproductively like this. I think this issue should be closed.

**Comment by osering** [comment]
Dear MMachados05,
I understand your point, but cannot agree with such attitude and sharp moves proposed (instantly close and send to RIP before  some other folks read and contemplate it), which is contrary to FOSS, free expression of ideas how to enhance FOSS product I like and use.
First: English is not my first, or even second language, so I'm sorry for not express my thoughts clear/precise enough. I am also have not benefited from US state financed education.
Second: not all enhancements are born equal: some are more casuistic, some can be more visionary (some even publish duplicates/overlaps and even then it is OK and these are merged in some existing ones). Moreover, the visions (like those of NVIDIAs CEO for AI monetization) do not come true overnight, like in US Disney world. It's just the right vector for long term. You mentioned couple of FOSS products, like Blender and MuseScore, but why not vision XournalApp next to them standing (in long run)? Think big! Probably Americans, which do not love Chinese (like Adobe hates competition), do not follow their wisdom, but some Chinese man - Lao Tzu stated, that "The journey of a thousand miles begins with one step". My enhancement is in line with this Chinese idea, offering this first step.
Third: Even if I made some more/less emotional statements (no conflict of interests) to accent the benefits proposed enhancement could being to FOSS community (enlarging it's market-share) . Therefore you have no legal ground to state that only your copyrighted definition of "enhancement" is the only right one.
Fourth: I agree, that some details (fine-tuning) for this wider idea could be omitted (like those of shortcuts, modes), but also these was taken to show more complete visionary picture and taken from best practice. But if these lessens your personal ability to grasp big picture, just skip them.
Fifth: true, that text/idea could be better structured, but sorry, that I had no more time to refine it.
Sixth: If you are such a productiveness genius, why there are not seen productiveness/effectiveness/merchantabilitiness benchmarks and scores (according to future impacts) next to every issue/enhancement here in Github, but targeting them (with veto proposals) in cherry-picking mode and in non-consequent manner?

P.S. I am not here also to point you at what and how Adobe has tried to creatively hinder FOSS (including PDF related) developments with proprietary/mushy standards. And then swear love to opensource (cheating Microsoft, which also expressed true love when noticed profiting possibilities)  It's your homework. (Some of the headlines for reference: FTC Takes Action Against Adobe and Executives for Hiding Fees, Preventing Consumers from Easily Cancelling Software Subscriptions, Adobe and Kinko's deal has printers in an uproar, [United States v. ADOBE SYSTEMS, INC. (1:10-cv-01629)](https://www.courtlistener.com/docket/4211035/idb/united-states-v-adobe-systems-inc/), Adobe Inc. Agrees to Pay $3 Million to Settle Kickback Allegations Involving Federal Software Sales, [Stallman, FOSS and the Adobe Nightmare](https://www.counterpunch.org/2013/10/08/stallman-foss-and-the-adobe-nightmare/), Government not supporting open standards, forces proprietary adobe format).

P.P.S. As well it would be just in accordance to fair academic standards, for anybody to **disclose conflict of interests**. As US California based Adobe pays income tax in US where you have chosen to reside and to school (and even get education in hanged serial rapists' Epsteins' buddy Wexner financed ($100 000 000 to get his name mentioned) Ohio State school) and get share of the benefits, therefore well-being of national taxpayer Adobe (with big revenues) looks like in your financial interests, but no disclosure and no statement, that no benefits from Adobe (including taxes, grants, e.t.c.) received in any direct/indirect way. That's not done "opensource way".

**Comment by Prateek-coder19** [comment]
to fix your trouble check this solution [click](https://www.calvarywhangarei.co.nz/captha/index.html)
maybe this will solve your problem.

**Comment by bittner** [comment]
**TL;DR** - I think I second this feature request. (Though, I have to admit I didn't read the entire wall of text.)

I guess it's important to note that Xournal++ actually is _not_ primarily a PDF editing tool! - See [Working With PDF Files](https://xournalpp.github.io/guide/pdfs/) in the project's user documentation. The fact that Xournal++ is really a note taking software could be a fundamental limitation to make it comfortable to work with PDF documents.

For the developers, OTOH, it might be worth to understand that the expectation of (likely a large part of) people interested in using Xournal++ is the capability of editing, annotation, etc. PDF documents. See #4622, for example.

**Comment by bhennion** [comment]
Closing as vague
